src/plugins/intel_gpu/src/graph/include/layout_optimizer.h

```diff
@@ -37,6 +37,32 @@ class primitive_inst;
 //  (note: layout_optimizer has internal caching mechanism, so if there's already reorder added for given (mem,format)
 //   pair during 'get_reorder' call, it will be reused).
 
+struct cache_key {
```

Comment:
I think we need to change the name. `cache_key` is now too general. maybe `reorder_cache_key`?